File: 프로그래머스/0/120923. 연속된 수의 합/연속된 수의 합.swift
import Foundation

func solution(_ num:Int, _ total:Int) -> [Int] {
    let mid = total / num + (num % 2 == 0 ? 1 : 0)
    let midIdx = num / 2
    
    return (0..<num).map { mid - midIdx + $0 }
}

File: 프로그래머스/0/120923. 연속된 수의 합/연속된 수의 합.js
function solution(num, total) {
    var isOdd = num % 2 != 0;
    var correction = isOdd ? 0 : 1;
    var mid = parseInt(total / num);
    var range = Math.floor(num / 2);
    var arr = []
    
    for (let i = 0; i < num; i++) {
        arr.push(mid - range + i + correction);
    }
    return arr;
}